apptools/preferences/preferences_helper.py

```diff
@@ -65,9 +72,23 @@ def _preferences_default(self):
     def _anytrait_changed(self, trait_name, old, new):
         """ Static trait change handler. """
 
+        if self.preferences is None:
+            return
+
+        # If the trait was a list or dict '_items' trait then just treat it as
+        # if the entire list or dict was changed.
+        if trait_name.endswith('_items'):
+            trait_name = trait_name[:-6]
+            if self._is_preference_trait(trait_name):
+                self.preferences.set(
+                    '%s.%s' % (self._get_path(), trait_name),
+                    getattr(self, trait_name)
+                )
+                return
```

Comment:
This is ugly, and is similar to what the UI counterpart is doing:
https://github.com/enthought/apptools/blob/83aa787c33edc220562e448ad649450f50bdffa0/apptools/preferences/ui/preferences_page.py#L76-L81

The alternative might have been a redesign of the helper <-> preference relationship, but that might not be feasible without changing how it is used.